### ISSUE TITLE
Diff loading stuck for certain HTML files

### ISSUE THREAD

**Issue by aleesaan** [microsoft/vscode#218674]
Type: <b>Bug</b>

Sometimes when I look at diffs for an HTML file it keeps loading forever, without actually showing the diff. It only happens for HTML files, never had this issue with other diffs. It happens both using the built-in git diff, and when looking at the diff via the gitlab extension.

VS Code version: Code 1.90.2 (Universal) (5437499feb04f7a586f677b155b039bc2b3669eb, 2024-06-18T22:37:41.291Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 (8 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.51GB free)|
|Process Argv|--crash-reporter-id c1340962-77d0-449f-81af-da82eb03d284|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|18.0.0
ng-switch|Bha|1.2.0
vscode-eslint|dba|3.0.10
gitlens|eam|15.1.0
copilot|Git|1.208.0
copilot-chat|Git|0.16.1
gitlab-workflow|Git|4.17.0
Angular2|joh|16.0.1
dotenv|mik|1.0.1
playwright|ms-|1.1.7
code-spell-checker|str|3.0.1
vscode-stylelint|sty|1.4.0
colonize|vms|2.2.2
change-case|wma|1.0.0
html-css-class-completion|Zig|1.20.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627cf:30244335
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelc:31048052
dsvsc021:30996838
f3je6385:31013174
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
legacy_priority:31082724

```

</details>

**Comment by hediet** [comment]
Can you share the html file diff that caused this problem?

**Comment by aleesaan** [comment]
Unfortunately I can't cause it's private company code :( but I know how annoyingly vague is this issue, so I'll keep you posted if I notice some sort of pattern that results in this infinite loading.

**Comment by hediet** [comment]
Closing, because without a repro this is unactionable. Moreover, a bug fix would not even be verifyable. Feel free to create a new issue once you can share more!

**Comment by aleesaan** [comment]
Hey @hediet I found out that disabling `diffEditor.experimental.showMoves` fixed the issue! So it looks like it has to do with that flag. 
Unfortunately as I said I can't share the exact code, but I noticed it happens in angular component templates, when adding an `ngIf` to wrap the whole file content.

For instance when I change something like this:

```html
a bunch of HTML content
```

to this:

```html
<ng-container *ngIf="someCondition; else content">
  new HTML content
</ng-container>
<ng-template #content>
  a bunch of HTML content
</ng-template>
```

So the `ng-container`/`ng-template` are the only additions, but the diff doesn't get it and just loads forever. Hope this can help you debug the issue as I really like the `showMoves` feature! For now I'll just keep it enabled and disable it whenever I need to open these templates that load forever  😬
